pkg/server/server.go

```diff
@@ -231,7 +252,17 @@ func (h *handler) authenticateEndpoint(w http.ResponseWriter, req *http.Request)
 	log = log.WithField("arn", identity.CanonicalARN)
 	var username string
 	var groups []string
-	if roleMapping, exists := h.lowercaseRoleMap[arnLower]; exists {
+	if nodeMapping, exists := h.lowercaseNodeMap[arnLower]; exists {
+		nodeName, err := h.nodeNameProvider.GetNodeName(identity.SessionName)
```

Comment:
If I understand this correctly, I can use instance-id as session name and gain access to my kubernetes cluster right?

Reply:
Only if you have the ability to assume the role that is in the nodeMapping.  Which is why it is critical that only ec2 be able to assume the roles you use for instances.